eduVPN/ConnectionViewController.swift

```diff
@@ -272,3 +286,31 @@ extension ConnectionViewController: Enter2FAViewControllerDelegate {
     }
     
 }
+
+extension ConnectionViewController: Enroll2FAViewControllerDelegate {
+    
+    func enroll2FA(controller: Enroll2FAViewController, didEnrollForType: TwoFactorType) {
+        // Fetch userInfo again so that connect method knows about twoFactor enrollment
+        ServiceContainer.providerService.fetchUserInfo(for: profile.info) { (result) in
+            DispatchQueue.main.async {
```

Comment:
Not sure if t his is different on macOS, but on iOS the convention is to call completion handlers on the main thread, instead of having the caller switch to the main thread.

Reply:
That's not the convention that Apple follows with `URLSession.dataTask(with ...`. I think there is something to say for either way. In this app I've had the caller side switch to the main thread where needed. Changing that throughout is not a trivial change, and only here would be inconsistent, hence I think it is best to leave it like this.

Reply:
Fair enough 👍 

Btw, you can choose on which thread those `URLSessionDataTask` handlers should be called by passing in a queue at initialization of the session. By default it might indeed use a background thread.